server/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -650,6 +651,7 @@ public SearchExecutionContext newSearchExecutionContext(
             shardId,
             shardRequestIndex,
             indexSettings,
+            clusterService != null ? clusterService.getClusterSettings() : ClusterSettings.createBuiltInClusterSettings(),
```

Comment:
Just curios, when is this null?

Reply:
Couldn't spot any, I was just defensive here. Removed the check.